src/manifolds/Segre.jl

```diff
@@ -0,0 +1,465 @@
+@doc raw"""
+    Segre{𝔽,V} <: AbstractManifold{𝔽}
+
+The Segre manifold
+````math
+    \mathcal{S} = \operatorname{Seg}(𝔽^{n_1} \times \dots \times 𝔽^{n_d})
+````
+is the set of rank-one tensors in ``𝔽^{n_1} \otimes \dots \otimes 𝔽^{n_d}``.
+
+When ``𝔽 = ℝ``, the Segre manifold is a normal Riemannian covering of
+````math
+    \mathcal{P} = ℝ^{+} \times S^{n_1 - 1} \times \dots \times S^{n_d - 1}
+````
+with the [warped product metric](https://en.wikipedia.org/wiki/Warped_product) [`inner`](@ref inner(::Segre, ::Any)). The tuple ``(n_1, \dots, n_d)`` is called the _valence_ of the manifold.
+
+The geometry is summarized in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+
+# Constructor
+    Segre(n::Int...; field::AbstractNumbers=ℝ)
+
+Generate a valence `(n, ...)` Segre manifold. Segre(n) is ``\mathbb{R} \setminus \{ 0 \}``.
+"""
+struct Segre{𝔽,V} <: AbstractManifold{𝔽} end
+
+function Segre(n::Int...; field::AbstractNumbers=ℝ)
+    return Segre{field,(n...,)}()
+end
+
+manifold_dimension(::Segre{𝔽,V}) where {𝔽,V} = (1 + sum(V .- 1))
+
+"""
+    check_size(M::Segre{𝔽, V}, p)
+
+Check whether `p` has the right size for `Segre` manifold `M`.
+"""
```

Comment:
Thi doc string (and the next four doc strings) does not provide any information.

We could either document properly what it means for `p` (or `X` for the next) to be of right size – or leave these two doc strings out. Overall this could even better be documented with `is_point` or `is_vector` doc strings, since these are the “API” functions a user would use. The `check_`s are merely the internal ones one should implement).

Yes I am writing a lot of comments – if you feel this gets too much work, just mention it, this one for example I know how I would do that as well.

Reply:
I'll just leave that docstring out then I think. But I think the docstring
```
    check_point(M::Segre{ℝ, V}, p; kwargs...)

Check whether `p` is a valid point on `M`, i.e. `p[1]` is a singleton containing a positive number and `p[i + 1]` is a point on `Sphere(V[i])`. The tolerance can be set using the `kwargs...`.
```
actually describes `is_point` rather than `check_point`. I'll put the docstring there as well.

I don't feel it gets too much. I use this library in my research and I have more manifolds I want to contribute so I don't mind :).

Reply:
Perfect! Exactly what I had in mind. One could also in general add a similar doc string to `is_vector`, that would be great


And: Great, that you do not mind and super cool that you use the library. I am very happy both about that and that you contribute. I think this PR is indeed nearly done.